frontend/src/context/CartContext.js

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useReducer } from 'react'
+import constate from 'constate'
+import logger from './Logger'
+import useLocalStorage from '../shared/useLocalStorage'
+
+const initialState = {
+  cart: [],
+  total: 0
+}
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'ADD_PRODUCT':
+      const id = action.payload.product.id
+      let addedProduct = state.cart.find(product => product.id === id)
+
+      if(addedProduct) {
+        addedProduct.quantity += 1
+
+        return {
+          ...state,
+          total: state.total + addedProduct.price
+        }
```

Comment:
Also, maybe I am blind, but are you not simply returning the old state without the by 1 increased quantity? Since you are not altering `state`anywhere and only total. Or does .find() return a reference?

Reply:
No you're definitely right. We should return an updated cart here. But I'm running into something annoying here that I can't manage to fix. The quantity is updated multiple times in one `ADD_PRODUCT` action, but the total isn't. I pushed an update, hopefully you can spot the problem quickly.

![image](https://user-images.githubusercontent.com/4533644/78522936-a2cf0c80-7801-11ea-9a8e-f8c530316386.png)

![image](https://user-images.githubusercontent.com/4533644/78522957-b7130980-7801-11ea-9c88-d2778ba969d1.png)